buildscripts/build.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+#
+# This script builds the application from source for multiple platforms.
+set -e
+
+# Get the parent directory of where this script is.
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ] ; do SOURCE="$(readlink "$SOURCE")"; done
+DIR="$( cd -P "$( dirname "$SOURCE" )/../" && pwd )"
+
+# Change into that directory
+cd "$DIR"
+
+# Get the git commit
+if [ -f $GOPATH/src/github.com/openebs/maya/GITCOMMIT ];
+then
+    GIT_COMMIT="$(cat $GOPATH/src/github.com/openebs/maya/GITCOMMIT)"
+else
+    GIT_COMMIT="$(git rev-parse HEAD)"
+fi
+
+# Get the version details
+VERSION="$(cat $GOPATH/src/github.com/openebs/maya/VERSION)"
+VERSION_META="$(cat $GOPATH/src/github.com/openebs/maya/BUILDMETA)"
+
+# Determine the arch/os combos we're building for
+UNAME=$(uname)
+ARCH=$(uname -m)
+if [ "$UNAME" != "Linux" -a "$UNAME" != "Darwin" ] ; then
+    echo "Sorry, this OS is not supported yet."
+    exit 1
+fi
+
+if [ "$UNAME" = "Darwin" ] ; then
+  XC_OS="darwin"
+elif [ "$UNAME" = "Linux" ] ; then
+  XC_OS="linux"
+fi
+
+if [ "${ARCH}" = "i686" ] ; then
+    XC_ARCH='386'
+elif [ "${ARCH}" = "x86_64" ] ; then
+    XC_ARCH='amd64'
+else
+    echo "Unusable architecture: ${ARCH}"
+    exit 1
+fi
+
+
+if [ -z "${PNAME}" ];
+then
+    echo "Project name not defined"
+    exit 1
+fi
+
+if [ -z "${CTLNAME}" ];
+then
+    echo "CTLNAME not defined"
+    exit 1
+fi
+
+# Delete the old dir
+echo "==> Removing old directory..."
+rm -rf bin/${PNAME}/*
+mkdir -p bin/${PNAME}/
+
+# If its dev mode, only build for ourself
+if [[ "${DEV}" ]]; then
+    XC_OS=$(go env GOOS)
+    XC_ARCH=$(go env GOARCH)
+fi
+
+# Build!
+echo "==> Building ${CTLNAME} using $(go version)... "
+
+GOOS="${XC_OS}"
+GOARCH="${XC_ARCH}"
+output_name="bin/${PNAME}/"$GOOS"_"$GOARCH"/"$CTLNAME
+
+if [ $GOOS = "windows" ]; then
+    output_name+='.exe'
+fi
+env GOOS=$GOOS GOARCH=$GOARCH go build -ldflags \
+    "-X github.com/openebs/maya/pkg/version.GitCommit=${GIT_COMMIT} \
+    -X main.CtlName='${CTLNAME}' \
+    -X github.com/openebs/maya/pkg/version.Version=${VERSION} \
+    -X github.com/openebs/maya/pkg/version.VersionMeta=${VERSION_META}"\
+    -o $output_name\
+    ./cmd/${CTLNAME}
+if [ "${PNAME}" = "cstor-volume-mgmt" ]; then
+    env GOOS=$GOOS GOARCH=$GOARCH go build -ldflags \
```

Comment:
Do we need GOOS and GOARCH, if we are only building the binary for the platform where we are running the buildscript from. Please correct me if my understanding is wrong.

Reply:
Yeah, we can directly use `XC_OS` and `XC_ARCH`, without declaring `GOOS` and `GOARCH`